### PR TITLE
Set up E2E test not to run on forked repositories

### PR DESCRIPTION
**Pull request by chengxiangdong**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -52,6 +52,8 @@ jobs:
       - name: make test
         run: make test
   e2e:
+    # This job does not run on forked repositories unless there is a self-runners.
```
Comment (RainbowMango) [inline .github/workflows/ci.yml:55]: The comment is not right. with this configuration, the job can't run even with self-runners.

Comment (chengxiangdong) [reply .github/workflows/ci.yml]: Oh yeah, that's an inappropriate comment

**Pull request merged**